### PR TITLE
Add tags values to Settings

### PR DESCRIPTION
**Pull request by maririos**
This PR is part of #93. It makes sure we can set and get settings with tags.
I will make a future PR to update tags.

### CODE REVIEW

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs [hunk 1]
```diff
@@ -33,6 +38,18 @@ private static void AssertEqual(ConfigurationSetting expected, ConfigurationSett
             Assert.AreEqual(expected.Label, actual.Label);
             Assert.AreEqual(expected.ContentType, actual.ContentType);
             Assert.AreEqual(expected.Locked, actual.Locked);
+            Assert.IsTrue(TagsEqual(expected.Tags, actual.Tags));
+        }
+
+        private static bool TagsEqual(IDictionary<string, string> expected, IDictionary<string, string> actual)
+        {
+            if (expected?.Count != actual?.Count) return false;
```
Comment (KrzysztofCwalina) [inline Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Will this work as you intended when both of the dictionaries are null?

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs [hunk 2]
```diff
@@ -33,6 +38,18 @@ private static void AssertEqual(ConfigurationSetting expected, ConfigurationSett
...
+            Assert.IsTrue(TagsEqual(expected.Tags, actual.Tags));
+        }
+
+        private static bool TagsEqual(IDictionary<string, string> expected, IDictionary<string, string> actual)
+        {
+            if (expected?.Count != actual?.Count) return false;
+            foreach (var pair in expected)
+            {
+                if (!actual.TryGetValue(pair.Key, out string value)) return false;
+                if (!string.Equals(value, pair.Value)) return false;
```
Comment (KrzysztofCwalina) [inline Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Our coding guidelines require that you use string.Equals overload that takes StringComparison enum.

File: Configuration/Azure.Configuration.Test/MockHttpClientTransport.cs [hunk 3]
```diff
@@ -301,6 +301,26 @@ bool WriteResponse(HttpResponseMessage response)
             return response.StatusCode == HttpStatusCode.OK;
         }
 
+        protected string GenerateExpectedRequestContent(ConfigurationSetting content)
+        {
+            string tags = "\"tags\":{";
```
Comment (KrzysztofCwalina) [inline Configuration/Azure.Configuration.Test/MockHttpClientTransport.cs]: You should either use StringBuilder or Utf8JsonWriter for building such complex Json payloads.

Comment (maririos) [reply Configuration/Azure.Configuration.Test/MockHttpClientTransport.cs]: Thank you. Changed!

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs (hunk 2 above)
Comment (maririos) [reply Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Changed!

File: Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs (hunk 1 above)
Comment (maririos) [reply Configuration/Azure.Configuration.Test/ConfigurationLiveTests.cs]: Hmm good point. 
The comparison itself works. The problem is after that, in the `foreach` where NullReferenceException happens.

**Pull request merged**